src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -198,6 +198,26 @@ int sd_dhcp6_client_set_duid(sd_dhcp6_client *client, uint16_t duid_type,
         return 0;
 }
 
+int sd_dhcp6_client_get_duid(sd_dhcp6_client *client, uint16_t *duid_type,
+                             const uint8_t **duid, size_t *duid_len) {
+        assert_return(client, -EINVAL);
+        assert_return(duid_type, -EINVAL);
+        assert_return(duid_len, -EINVAL);
+        assert_return(duid, -EINVAL);
```

Comment:
In functions which allow multiple parameters to be queried, it is nice to be able to query just a subset sometimes. For example it's certainly possible that the caller might only need to now the duid_type, without needing the value. Similarly for other parameters. A lot of systemd functions are like that.